### PR TITLE
Fix: Move EM and Diagrams API to beta client

### PR DESCRIPTION
**Pull request by marvrez**
## Description
This work fixes the  tests for the entity matcher and the diagrams api. Many of the endpoints in the mentioned APIs are in beta and since the Context API now actually check for `cdf-version` in the header, an error will be returned if the corresponding header value is not `beta`. The fix is to therefore use the beta client in the EM and diagrams integration tests.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change. 
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring. 
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/_cognite_client.py [hunk 1]
```diff
@@ -62,15 +62,15 @@ def __init__(self, config: Optional[ClientConfig] = None) -> None:
...
+        self.entity_matching = EntityMatchingAPI(self._config, api_version="beta", cognite_client=self)
         self.templates = TemplatesAPI(self._config, api_version=self._API_VERSION, cognite_client=self)
         self.vision = VisionAPI(self._config, api_version=self._API_VERSION, cognite_client=self)
         self.extraction_pipelines = ExtractionPipelinesAPI(self._config, api_version="playground", cognite_client=self)
         self.extraction_pipeline_runs = ExtractionPipelineRunsAPI(
             self._config, api_version="playground", cognite_client=self
         )
         self.transformations = TransformationsAPI(self._config, api_version=self._API_VERSION, cognite_client=self)
-        self.diagrams = DiagramsAPI(self._config, api_version=self._API_VERSION, cognite_client=self)
+        self.diagrams = DiagramsAPI(self._config, api_version="beta", cognite_client=self)
```
Comment (erlendvollset) [inline cognite/client/_cognite_client.py:73]: beta APIs should only be made available through the cognite.client.beta.CogniteClient class.

Comment (marvrez) [reply cognite/client/_cognite_client.py]: https://github.com/cognitedata/cognite-sdk-python/pull/1029/commits/d6d62412e7e44915e848a30ed487157446347709

File: tests/tests_integration/conftest.py [hunk 2]
```diff
@@ -4,21 +4,21 @@
...
 
 from cognite.client import ClientConfig, CogniteClient
-from cognite.client.credentials import OAuthClientCredentials, OAuthInteractive
+from cognite.client.beta import CogniteClient as CogniteBetaClient
+from cognite.client.credentials import CredentialProvider, OAuthClientCredentials, OAuthInteractive
 
 
-@pytest.fixture(scope="session")
-def cognite_client() -> CogniteClient:
+def _get_credentials() -> CredentialProvider:
```
Comment (erlendvollset) [inline tests/tests_integration/conftest.py:11]: Let's make this a session-scoped fixture aswell so we don't instantiate the credential providers multiple times (the credential provider instantiation has side effects).

**Review by erlendvollset — APPROVED** [review]
You also need to move the APIs into the beta client, so they're not available through the main client.

**Review by erlendvollset — CHANGES_REQUESTED** [review]
hit approve by mistake

File: tests/tests_integration/conftest.py (hunk 2 above)
Comment (marvrez) [reply tests/tests_integration/conftest.py:12]: Tried it in this commit https://github.com/cognitedata/cognite-sdk-python/pull/1029/commits/3c9f48809dad6ab1d527acb323ef6a4b6d703085, seems like it doesn't work

**Comment by marvrez** [comment]
Moved the APIs to the beta client in https://github.com/cognitedata/cognite-sdk-python/pull/1029/commits/50dbda317c5d49c681986b37633acb49dc5b8480. Thanks for the suggestion.

**Comment by marvrez** [comment]
Closing this PR as the fix is instead implemented in the Context API. The fix is to revert to the previous logic.

**Pull request closed without merging**